app/src/androidTest/java/org/mozilla/focus/activity/DownloadFileTest.kt

```diff
@@ -135,4 +140,78 @@ class DownloadFileTest {
             assertNativeAppOpens(GOOGLE_PHOTOS)
         }
     }
+
+    @SmokeTest
+    @Test
+    fun downloadAndOpenPDFFileTest() {
+        downloadFileName = "washington.pdf"
+
+        searchScreen {
+        }.loadPage(downloadTestPage) {
+            progressBar.waitUntilGone(waitingTime)
+        }
+        browserScreen {
+            clickLinkMatchingText(downloadFileName)
+        }
+        // If permission dialog appears on devices with API<30, grant it
+        if (permAllowBtn.waitForExists(waitingTime)) {
+            permAllowBtn.click()
+        }
+        downloadRobot {
+            verifyDownloadDialog(downloadFileName)
+            clickDownloadButton()
+            verifyDownloadConfirmationMessage(downloadFileName)
+            openDownloadedFile()
+            assertNativeAppOpens(GOOGLE_DRIVE)
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun downloadAndOpenWEBMFileTest() {
+        downloadFileName = "videoSample.webm"
+
+        searchScreen {
+        }.loadPage(downloadTestPage) {
+            progressBar.waitUntilGone(waitingTime)
+        }
+        browserScreen {
+            clickLinkMatchingText(downloadFileName)
+        }
+        // If permission dialog appears on devices with API<30, grant it
+        if (permAllowBtn.waitForExists(waitingTime)) {
+            permAllowBtn.click()
+        }
+        downloadRobot {
+            verifyDownloadDialog(downloadFileName)
+            clickDownloadButton()
+            verifyDownloadConfirmationMessage(downloadFileName)
+            openDownloadedFile()
+            assertNativeAppOpens(GOOGLE_PHOTOS)
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun verifyDownloadedFileOnStorageTest() {
+        downloadFileName = "executable.exe"
```

Comment:
An exe file is not really important on Android, I would use a more commonly used type of file: apk, mp3, xls, etc

Reply:
Done!